pkg/networkserver/mac/link_adr.go

```diff
@@ -322,16 +322,23 @@ func HandleLinkADRAns(
 	}
 
 	ev := EvtReceiveLinkADRAccept
+	rejected := false
 
 	// LoRaWAN 1.0.4 spec L534-538:
 	// An end-device SHOULD accept the channel mask controls present in LinkADRReq, even
 	// when the ADR bit is not set. The end-device SHALL respond to all LinkADRReq commands
 	// with a LinkADRAns indicating which command elements were accepted and which were
 	// rejected. This behavior differs from when the uplink ADR bit is set, in which case the end-
 	// device accepts or rejects the entire command.
-	if (!adrEnabled && !pld.ChannelMaskAck) ||
-		(adrEnabled && !pld.DataRateIndexAck) ||
-		(adrEnabled && !pld.TxPowerIndexAck) {
+	if macspec.UseADRBit(macState.LorawanVersion) {
+		rejected = (!adrEnabled && !pld.ChannelMaskAck) ||
+			(adrEnabled && !pld.DataRateIndexAck) ||
+			(adrEnabled && !pld.TxPowerIndexAck)
```

Comment:
Is this the correct boolean logic? Should the first `!adrEnabled` be with negate operator?